flow/model/model.go

```diff
@@ -63,19 +63,19 @@ type Record interface {
 // encoding/gob cannot encode unexported fields
 type RecordItems struct {
 	ColToValIdx map[string]int
-	Values      []*qvalue.QValue
+	Values      []qvalue.QValue
 }
 
 func NewRecordItems() *RecordItems {
 	return &RecordItems{
 		ColToValIdx: make(map[string]int),
 		// create a slice of 32 qvalues so that we don't have to allocate memory
 		// for each record to reduce GC pressure
-		Values: make([]*qvalue.QValue, 0, 32),
+		Values: make([]qvalue.QValue, 0, 32),
```

Comment:
This is most worrisome part, where we go from preallocating 256 bytes to 1024 bytes. Given this PR's goal is to reduce overall GC pressure, it might make sense to dial this preallocation down to 8 or 16 elements as a compromise

Reply:
Looking over with Kaushik, we worked out that it's straight forward to have the length, so no need to pick a hopeful capacity